### ISSUE TITLE
misleading info given by `Pkg.status()`

### ISSUE THREAD

**Issue by ThomasBreuer** [JuliaLang/Pkg.jl#2615]
The following happens with Julia 1.5.3.
My `DEPOT_PATH` consists of two paths, the one to the `.julia` subdirectory of my home directory and another one, and the latter is active.
With `Pkg.status()`, I get a list of packages. One of them is marked with an arrow, and an info message says

> packages marked with → not downloaded, use `instantiate` to download

However, the package in question can be loaded without problem.
When I try (just for curiosity) `instantiate` then apparently some artifact gets downloaded to `/tmp` but otherwise nothing changes.
I must admit that `~/.julia/artifacts/Overrides.toml` contains a redirection for the package in question, and the path stored there is not valid anymore (and the problem disappears when I correct this path), but why does `Pkg.status()` look into this place, which is not relevant for the active environment?

**Comment by KristofferC** [comment]
My guess is that https://github.com/JuliaLang/Pkg.jl/blob/0de93211d96c57c1617000a683b70a70154439d5/src/Operations.jl#L622-L632 returns false which it should return true.

It could be https://github.com/JuliaLang/julia/blob/e35028acd307db6c33cf857c5faa1da497d197ad/stdlib/Artifacts/src/Artifacts.jl#L252-L254 that is not correctly finding the artifact. You could try calling it manually.

You could dig down a bit into those functions and see if you find something.

**Comment by ThomasBreuer** [comment]
@KristofferC Thanks for the hint.
`Pkg.Artifacts.artifact_paths` returns an array of length two (corresponding to the two paths in `DEPOT_PATH`) if `honor_overrides` is `false`, and in the `true` case it returns an array of length one, corresponding to the entry in `~/.julia/artifacts/Overrides.toml`; this entry is an invalid path.
This means that this `Overrides.toml` file overrides the settings for *all* environments, not just for the one where the file resides (the currently inactive `~/.julia`).
This seems to be not consistent with the behaviour of `using`, since this command finds the artifact in the path of the active environment (without override).

**Comment by KristofferC** [comment]
I agree that they should agree with each other. I think this makes it an issue for the Julia repo then since it seems like that is where the disagreement is?
